### PR TITLE
Add error message checks

### PR DESCRIPTION
**Pull request by banasraf**
Signed-off-by: Rafal <Banas.Rafal97@gmail.com>

## Description
- [ ] **Bug fix** (*non-breaking change which fixes an issue*)
- [ ] **New feature** (*non-breaking change which adds functionality*)
- [ ] **Breaking change** (*fix or feature that would cause existing functionality to not work as expected*)
- [ ] **Refactoring** (*Redesign of existing code that doesn't affect functionality*)
- [x] **Other** (*e.g. Documentation, Tests, Configuration*)

### What happened in this PR

Adding error message checks + some test fixes
#### Additional information
- Affected modules and functionalities:

- Key points relevant for the review:

### Tests
- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: dali/test/python/test_dali_tf_dataset_eager.py [hunk 1]
```diff
@@ -265,17 +288,27 @@ def check_disallowed_es(kwargs, input_datasets):
...
+def test_tf_dataset_es_with_source():
     in_dataset = tf.data.Dataset.from_tensors(np.full((2, 2), 42)).repeat()
-    # no names
-    yield check_disallowed_es, {}, {}
-    # num_outputs
-    yield check_disallowed_es, {'name': 'a', 'num_outputs': 1}, {'a': in_dataset}
-    # source provided, so we don't have valid placeholder
-    yield check_disallowed_es, {'name': 'a', 'source': []}, {'a': in_dataset}
-    # misnamed placeholder
-    yield check_disallowed_es, {'name': 'b'}, {'a': in_dataset}
```
Comment (banasraf) [inline dali/test/python/test_dali_tf_dataset_eager.py:278]: This one is already covered on L260

File: dali/test/python/test_dali_tf_dataset_eager.py [hunk 2]
```diff
@@ -178,44 +183,44 @@ def test_tf_dataset_multi_input():
...
     batch_size = 12
     num_threads = 4
-    iterations = 10
 
     pipeline = get_image_pipeline(batch_size, num_threads, 'gpu', 0)
 
     with tf.device('/cpu:0'):
-        dataset = get_dali_dataset_from_pipeline(
-            pipeline, batch_size, num_threads, 'cpu', 0)
+        dataset = get_dali_dataset_from_pipeline(pipeline, 'cpu', 0)
```
Comment (JanuszL) [inline dali/test/python/test_dali_tf_dataset_eager.py:210]: 👍

File: dali/test/python/test_dali_tf_dataset_eager.py [hunk 3]
```diff
@@ -236,22 +248,33 @@ def test_tf_dataset_wrong_input_type():
...
+
 
 @pipeline_def(batch_size=10, num_threads=4, device_id=0)
 def es_pipe(kwargs):
     return fn.external_source(**kwargs)
 
 
-@raises(Exception)
-def check_disallowed_es(kwargs, input_datasets):
+# @raises(Exception)
```
Comment (JanuszL) [inline dali/test/python/test_dali_tf_dataset_eager.py:276]: ?

File: dali/test/python/test_dali_tf_dataset_eager.py [hunk 4]
```diff
@@ -35,7 +37,10 @@ def test_tf_dataset_cpu():
     run_tf_dataset_eager_mode('cpu')
 
 # Return differently sized images to check if DALIDataset can handle this case gracefully
-@raises(tf.errors.FailedPreconditionError)
+@raises(tf.errors.FailedPreconditionError, glob='*Batch output at index * from DALI pipeline is not uniform - individual '
```
Comment (JanuszL) [inline dali/test/python/test_dali_tf_dataset_eager.py:40]: You can remove leading and trailing `*` from glob pattern. They are not needed, as glob to regexp converted we use adds them automatically.

File: dali/test/python/test_functional_api.py [hunk 5]
```diff
@@ -45,11 +43,13 @@ def _test_fn_rotate(device):
...
     assert(np.array_equal(arr, ref))
 
+@raises(TypeError, glob='Illegal pipeline output type. The output * contains a nested'
+                        ' `DataNode`. Missing list/tuple expansion (*) is the likely cause.')
 def test_set_outputs():
     data = [[[np.random.rand(1, 3, 2)], [np.random.rand(1, 4, 5)]]]
     pipe = Pipeline(batch_size = 1, num_threads = 1, device_id = None)
     pipe.set_outputs(fn.external_source(data, num_outputs = 2, cycle = 'quiet'))
-    assert_raises(TypeError, pipe.build)
+    pipe.build()
```
Comment (JanuszL) [inline dali/test/python/test_functional_api.py:52]: I would stick to checking if the `build` step raises that error, not any other one.

File: dali/test/python/test_operator_pad.py [hunk 6]
```diff
@@ -12,6 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from nose.tools.nontrivial import raises
```
Comment (JanuszL) [inline dali/test/python/test_operator_pad.py:15]: ?

File: dali/test/python/test_operator_pad.py [hunk 7]
```diff
@@ -150,7 +151,7 @@ def test_pad_error():
                                 align=align, shape_arg=shape_arg, layout=layout)
 
     pipe.build()
-    assert_raises(RuntimeError, pipe.run)
+    pipe.run()
```
Comment (JanuszL) [inline dali/test/python/test_operator_pad.py:154]: I would keep this check here.

File: dali/test/python/test_video_pipeline.py [hunk 8]
```diff
@@ -90,9 +90,11 @@ def test_simple_videopipeline():
         assert(out[0].layout() == "FHWC")
     del pipe
 
+@raises(RuntimeError, glob='*There are no valid sequences in the provided dataset, '
+                           'check the length of the available videos and the requested sequence length')
 def test_wrong_length_sequence_videopipeline():
     pipe = VideoPipe(batch_size=BATCH_SIZE, data=VIDEO_FILES, sequence_length=100000)
-    assert_raises(RuntimeError, pipe.build)
+    pipe.build()
```
Comment (JanuszL) [inline dali/test/python/test_video_pipeline.py:97]: Can we keep this check here?

File: dali/test/python/test_video_pipeline.py [hunk 9]
```diff
@@ -216,19 +223,32 @@ def _test_file_list_empty_videopipeline(start, end):
...
+    with assert_raises(RuntimeError, glob='*Start frame number should be lesser or equal to end frame number for a file*'):
+        pipe.build()
     os.remove(list_file.name)
 
-def test_file_list_empty_videopipeline():
+
+def test_file_list_invalid_range():
     invalid_ranges = [
-        [0, 0],
-        [10, 10],
```
Comment (JanuszL) [inline dali/test/python/test_video_pipeline.py:225]: Why you removed this cases?

File: dali/test/python/test_video_pipeline.py [hunk 10]
```diff
@@ -216,19 +223,32 @@ def _test_file_list_empty_videopipeline(start, end):
...
+
+
+def test_file_list_empty():
+    files = sorted(os.listdir(VIDEO_DIRECTORY))
+    list_file = tempfile.NamedTemporaryFile(mode="w", delete=False)
+    list_file.write("{} {} {} {}\n".format(os.path.join(VIDEO_DIRECTORY, files[0]), 0, 10, 10))
+    list_file.close()
+
+    pipe = VideoPipeList(batch_size=BATCH_SIZE, data=list_file.name)
+    with assert_raises(RuntimeError, glob='*No files were read'):
```
Comment (JanuszL) [inline dali/test/python/test_video_pipeline.py:248]: I would improve video_loader error message in this case as we discussed offline.

File: dali/test/python/test_video_pipeline.py [hunk 11]
```diff
@@ -216,19 +223,32 @@ def _test_file_list_empty_videopipeline(start, end):
...
         [-1, 1],
         [1000000, None],
         [0, -1000]
     ]
     for r in invalid_ranges:
-        yield _test_file_list_empty_videopipeline, r[0], r[1]
+        yield _test_file_list_invalid_range, r[0], r[1]
+
+
+def test_file_list_empty():
```
Comment (JanuszL) [inline dali/test/python/test_video_pipeline.py:241]:
```suggestion
def test_file_list_empty_range():
```

File: dali/test/python/test_dali_tf_dataset_eager.py (hunk 4 above)
Comment (mzient) [reply dali/test/python/test_dali_tf_dataset_eager.py:40]: You can skip most of the message and only match relevant fragments, making some slack for future adjustment of the error messages.

File: dali/test/python/test_dali_tf_dataset_eager.py [hunk 12]
```diff
@@ -224,6 +229,13 @@ def check_tf_dataset_wrong_input_type(wrong_input_datasets):
                 device_id=pipe.device_id)
         return dali_dataset
 
+
+@raises(TypeError, glob='*`input_datasets` must be a dictionary that maps input names (the `name` specified for External '
+                        'Source node in DALI pipeline) to input datasets objects (`tf.data.Dataset`) or '
+                        '`nvidia.dali.plugin.tf.experimental.Input` wrapper objects*')
```
Comment (mzient) [inline dali/test/python/test_dali_tf_dataset_eager.py:235]: Likewise, you don't have to put the entire message here - and you can skip leading and trailing *.

File: dali/test/python/test_operator_pad.py (hunk 6 above)
Comment (mzient) [reply dali/test/python/test_operator_pad.py:15]: Please remove. That's probably VSCode's doing - it keeps doing it to me all the time :\

File: dali/test/python/test_dali_tf_dataset_eager.py (hunk 3 above)
Comment (banasraf) [reply dali/test/python/test_dali_tf_dataset_eager.py]: leftover - removed

File: dali/test/python/test_dali_tf_dataset_eager.py (hunk 4 above)
Comment (banasraf) [reply dali/test/python/test_dali_tf_dataset_eager.py]: I removed all the leading and trailing * and shortened the patterns to the essential parts.

File: dali/test/python/test_functional_api.py (hunk 5 above)
Comment (banasraf) [reply dali/test/python/test_functional_api.py]: done

File: dali/test/python/test_operator_pad.py (hunk 6 above)
Comment (banasraf) [reply dali/test/python/test_operator_pad.py]: removed

File: dali/test/python/test_operator_pad.py (hunk 7 above)
Comment (banasraf) [reply dali/test/python/test_operator_pad.py]: done

File: dali/test/python/test_video_pipeline.py (hunk 8 above)
Comment (banasraf) [reply dali/test/python/test_video_pipeline.py]: done

File: dali/test/python/test_video_pipeline.py (hunk 11 above)
Comment (banasraf) [reply dali/test/python/test_video_pipeline.py]: done

File: dali/test/python/test_dali_tf_dataset_eager.py (hunk 12 above)
Comment (banasraf) [reply dali/test/python/test_dali_tf_dataset_eager.py]: I shortened the patterns to the essential part

File: dali/test/python/test_video_pipeline.py (hunk 11 above)
Comment (banasraf) [reply dali/test/python/test_video_pipeline.py]: done

File: dali/test/python/test_video_pipeline.py (hunk 10 above)
Comment (banasraf) [reply dali/test/python/test_video_pipeline.py]: As we agreed, the warning  about empty range, that is already there,  should be enough

**Review by JanuszL — APPROVED** [review]

**Review by awolant — APPROVED** [review]

**Comment by banasraf** [comment]
!build

**Pull request merged**